nixos/modules/flyingcircus/roles/statshost.nix

```diff
@@ -89,24 +101,57 @@ in
       rootUrl = "http://${cfg.roles.statshost.hostName}/grafana";
     };
 
-    services.nginx.enable = true;
-    services.nginx.httpConfig = ''
-      server {
-          listen *:80;
-          server_name ${cfg.roles.statshost.hostName};
-
-          location / {
-              rewrite . /grafana/ redirect;
-          }
-
-          location /grafana/ {
-              proxy_pass http://localhost:3001/;
-          }
-
-          location /grafana/public {
-              alias ${config.services.grafana.staticRootPath};
-          }
-      }
+    flyingcircus.roles.nginx.enable = true;
+
+    flyingcircus.roles.nginx.httpConfig = let
+      httpHost = cfg.roles.statshost.hostName;
+      in
+      if cfg.roles.statshost.useSSL then ''
+        server {
+            listen *:80;
+            server_name ${httpHost};
+            rewrite . https://$server_name$request_uri redirect;
+        }
+
+        server {
+            listen *:443 ssl;
+            server_name ${httpHost};
+
+            ssl_certificate ${/etc/local/nginx/stats.crt};
+            ssl_certificate_key ${/etc/local/nginx/stats.key};
+
+            # add_header Strict-Transport-Security "max-age=31536000";
+
+            location / {
+                rewrite . /grafana/ redirect;
+            }
+
+            location /grafana/ {
+                proxy_pass http://localhost:3001/;
+            }
+
+            location /grafana/public {
+                alias ${config.services.grafana.staticRootPath};
+            }
+        }
+      '' else
+      ''
+        server {
```

Comment:
Don't repeat yourself! It's just a matter of time when someone performs a change on only on of both "location" sets. Going to fix it by myself due to EuroPython.